woocommerce/woocommerce-ios
# Make Blaze section on Dashboard dismissable

Closes: #11302 


## What

This PR adds a new button on the Blaze section on the My Store screen. This button displays only one option, which is to dismiss the section for the currently selected store. When the user creates a Blaze campaign for the same store, the section will reappear.

## How
- Added a new `UserDefaults` key `hasDismissedBlazeSectionOnMyStore` to keep track of the state of the section on the My Store screen.
- Added an extension of `UserDefaults` to get, set and reset the values for `hasDismissedBlazeSectionOnMyStore`.
- Updated `BlazeWebViewModel` to trigger restoring the Blaze section upon completion of campaign creation.
- Updated `BlazeCampaignDashboardView` to show a button for dismissing the section.
- Updated `BlazeCampaignDashboardViewModel` to handle dismissing the section and observing the state of its visibility.

## Testing instructions

- Log in to a store that is eligible for creating Blaze campaigns.
- Notice that the Blaze section is initially displayed below the Stats section, with an ellipsis menu on the top right.
- Tap the ellipsis button, notice that an option to hide the section is displayed.
- Tap Hide Blaze, the section should be dismissed. Confirm that the following event is tracked: `blaze_view_dismissed`, Properties: {"source":"my_store_section"}.
- Relaunch the app, the Blaze section should still be hidden. 
- Switch to another store eligible for Blaze if you account has access to one. Notice that the Blaze section is still available for this store.
- Switch back to the initial store and create a Blaze campaign. After completion, the Blaze campaign should be displayed again on the My Store screen.
- Dismiss the section and again and log out of your account.
- Log back in, the Blaze section should be visible on the My Store screen.

## Screenshots


<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/5700782c-015f-43a1-ae95-fef6f68fe534" width=320 /> <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/2b6d0f18-706f-4eec-b6c8-1bc2129ea815" width=320 />



---
